CMake/SetCompilerFlags.cmake

```diff
@@ -0,0 +1,36 @@
+if(MSVC)
```

Comment:
What is this file and how is it different from `SetFlags.cmake`?

Reply:
> "Who are you and what have you done with the real SetFlags?" peterbell angrily demands of SetCompilerFlags.

It's a migration of SetFlags that removes functionality now handled natively by CMake and to use targets so we don't have to patch around compiler warnings generated by library code. The original SetFlags was only renamed to avoid polluting the diff; after this is merged I'm gonna do a second round finishing everything up.

Reply:
So SetFlags isn't actually used any more? We should keep the same warning level, not just remove the flags.

Reply:
You also seem to missing link-time optimisation flags on MSVC.

Reply:
Agreed, right now Wall/extra is enabled but warnings-as-errors isn't yet, I am intending to fix it in a follow up PR alongside PCH (target_precompile_headers) on Tools/.

I can confirm that Whole Program Optimisation on Visual Studio was enabled by setting CMAKE_INTERPROCEDURAL_OPTIMISATION.